kie-server-parent/kie-server-services/kie-server-services-common/src/test/java/org/kie/server/services/impl/KieServerImplInitTest.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.server.services.impl;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.kie.server.api.KieServerConstants;
+import org.kie.server.api.KieServerEnvironment;
+import org.kie.server.api.model.KieContainerResource;
+import org.kie.server.api.model.KieContainerStatus;
+import org.kie.server.api.model.KieServerInfo;
+import org.kie.server.api.model.ReleaseId;
+import org.kie.server.controller.api.KieServerController;
+import org.kie.server.controller.api.model.KieServerSetup;
+import org.kie.server.services.api.StartupStrategy;
+import org.kie.server.services.impl.storage.KieServerState;
+import org.kie.server.services.impl.storage.file.KieServerStateFileRepository;
+
+public class KieServerImplInitTest {
+
+    private static final File REPOSITORY_DIR = new File("target/repository-dir");
+    private static final String KIE_SERVER_ID = "kie-server-impl-test";
+
+    private KieServerImpl kieServer;
+    private String origServerId = null;
+
+    @Before
+    public void setupKieServerImpl() throws Exception {
+        origServerId = KieServerEnvironment.getServerId();
+        System.setProperty("org.kie.server.id", KIE_SERVER_ID);
+        KieServerEnvironment.setServerId(KIE_SERVER_ID);
+
+        FileUtils.deleteDirectory(REPOSITORY_DIR);
+        FileUtils.forceMkdir(REPOSITORY_DIR);                
+    }
+
+    @After
+    public void cleanUp() {
+        if (kieServer != null) {
+            kieServer.destroy();
+        }
+        KieServerEnvironment.setServerId(origServerId);
+        System.clearProperty(KieServerConstants.KIE_SERVER_STARTUP_STRATEGY);
+        StartupStrategyProvider.clear();
+    }
+    
+    @Test
+    public void testStartupStrategyProviderDefault() {
+        StartupStrategy strategy = StartupStrategyProvider.get().getStrategy();
+        assertTrue(strategy instanceof ControllerBasedStartupStrategy);
+    }
+    
+    @Test
+    public void testStartupStrategyProviderLocalContainers() {        
+        System.setProperty(KieServerConstants.KIE_SERVER_STARTUP_STRATEGY, LocalContainersStartupStrategy.class.getSimpleName());
```

Comment:
Will this change anything? Since here https://github.com/kiegroup/droolsjbpm-integration/pull/1422/files#diff-321935d038fba590140afb75dcc87e9dR35 it is set only once when a class is loaded.

Reply:
See next line that on clear will actually recreate StartupStrategyProvider instance

Reply:
Ah, sorry, I thought it was a static field. My bad.